server/methods.js

```diff
@@ -164,17 +170,34 @@ function createProductImage(product) {
   });
 }
 
+async function createProductImageFromUrl(product) {
+  const url = Promise.await(randomPuppy());
```

Comment:
Once you are in an `async` function, you should be able to use the normal `await` rather than `Promise.await`